File: advent_of_code/2022/day1/src/part1.rs
pub fn solve(input: &str) -> i32 {
    let lines = input.lines();

    let mut max = 0;

    let mut current = 0;
    for line in lines {
        if line.is_empty() {
            max = max.max(current);
            current = 0;
            continue;
        }

        current += line
            .parse::<i32>()
            .expect("The line doesn't contain a number");
    }

    max
}

File: advent_of_code/2022/day1/zig/src/part1.zig
const std = @import("std");
pub fn solve(input: []u8) !i32 {
    var lines = std.mem.split(u8, input, "\n");

    var max: i32 = 0;
    var current: i32 = 0;
    while (lines.next()) |line| {
        if (line.len == 0) {
            max = @max(max, current);
            current = 0;
            continue;
        }

        current += try std.fmt.parseInt(i32, line, 10);
    }

    return max;
}
